django/template/context.py

```diff
@@ -251,7 +251,14 @@ def bind_template(self, template):
         processors = template.engine.template_context_processors + self._processors
         updates = {}
         for processor in processors:
-            updates.update(processor(self.request))
+            try:
+                updates.update(processor(self.request))
```

Comment:
What about moving `processor(self.request)` outside the try/except so the exception isn't wrapped if `processor(self.request)` raises `TypeError`?

Reply:
Good point